src/main/java/com/tjtanjin/steve/tasks/TaskHandler.java

```diff
@@ -0,0 +1,162 @@
+                taskName, "incomplete", taskType.toUpperCase(), taskDates)) {
+            return "Info: Your task has been added! You now have " + TASKS.size() + " task(s)!";
+        } else {
+            return "Error: There was an error saving your task, please check your syntax and try again.";
+        }
+    }
+
+    /**
+     * Gets array list of tasks.
+     *
+     * @return array list of tasks
+     */
+    public ArrayList<Task> getTasks() {
+        return TASKS;
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param index index of task to mark as done
+     * @return string response after operation is done
+     */
+    public String markTaskDone(int index) {
+        try {
+            Task task = TASKS.get(index);
+            if (task.getStatus().equals("complete")) {
+                return "Error: Task is already completed!";
+            } else {
+                task.markCompleted();
+                String taskName = task.getTaskName();
+                String taskType = task.getType().toUpperCase();
+                LocalDate[] taskDates = task.getDates();
+                if (STORAGE_HANDLER.saveTask(index, "DONE",
+                        taskName, "complete", taskType, taskDates)) {
+                    return "Info: Yay your task is done! :D";
+                } else {
+                    return "Error: There was an error marking your task as done, "
+                            + "please check your syntax and try again.";
+                }
+            }
+        } catch (IndexOutOfBoundsException e) {
+            return "Error: The specified task index does not exist!";
+        }
+    }
+
+    /**
+     * Deletes a task.
+     *
+     * @param index index of task to delete
+     * @return string response after operation is done
+     */
+    public String deleteTask(int index) {
+        try {
+            Task task = TASKS.get(index);
+            TASKS.remove(task);
+            String taskName = task.getTaskName();
+            String taskType = task.getType().toUpperCase();
+            LocalDate[] taskDates = task.getDates();
+            if (STORAGE_HANDLER.saveTask(index, "DELETE",
+                    taskName, "complete", taskType, taskDates)) {
+                return "Info: The following task has been deleted:\n" + task;
+            } else {
+                return "Error: There was an error deleting your task, "
+                        + "please check your syntax and try again.";
+            }
+        } catch (IndexOutOfBoundsException e) {
+            return "Error: The specified task index does not exist!";
+        }
+    }
+
+    /**
+     * Shows all tasks to user
+     *
+     * @return string response after operation is done
+     */
+    public String showAllTasks() {
+        if (TASKS.size() == 0) {
+            return "Info: You have no task at the moment!";
+        } else {
+            StringBuilder str = new StringBuilder("Info: ");
```

Comment:
Perhaps rename str to something more descriptive like stringOfAllTasks.

Reply:
Fixed as with the previous comment :D Thank you!